HotelBookingTests/HotelExampleBookRoomCantFailTest.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System;
+using HotelBase;
+using HotelBooking;
+using NUnit.Framework;
+using Rhino.Mocks;
+
+namespace HotelBookingTests
+{
+    [TestFixture]
+    public class HotelExampleBookRoomCantFailTest : MockedServices
+    {
+        private HotelExampleBookRoomCantFail _hotel;
+
+        [SetUp]
+        public void Setup()
+        {
+            _hotel = new HotelExampleBookRoomCantFail(BookingService, PaymentService, Logger);
+        }
+
+        [Test]
+        public void ShouldFailReservationResultIfBookRoomFails()
+        {
+            string email = "test@test.com";
+            DateTime date = DateTime.Parse("01.01.2020");
+            int creditCardNumber = 1234567;
+            double price = 200;
+            BookingService.Stub(x => x.Book(date)).Return(false).Repeat.Once();
+            PaymentService.Stub(x => x.Pay(creditCardNumber, price)).Return(true).Repeat.Once();
+
+            ReservationResult result = _hotel.Reserve(date, price, creditCardNumber, email);
+
+            Assert.False(result.Success);
```

Comment:
Should have:

this should be the only assert in this test. You want to test scenario when hotel booking faild  and as a result whole reservation is not success.  Other asserts are less important. 

Now , I know why you assert for other results and yes for such a classs as you wrote you should test other results, maybe not in one test, but still. What is more important is that this test is trying to tell you something. The same thing you had to test in other Hotel classes and here again. Test trying to tell you that maybe it will be good to to have just one code, which will "execute booking" and aggregate all steps results. 

This comment is related to comment in Hotel class.

Reply:
I'm not sure if I get the idea here. Anyway I've added the asserts for reservation steps in base hotel test and removed them from specific hotel tests.